scripts/linkValidator.js

```diff
@@ -1,59 +1,107 @@
-  const errors = [];
+/** @type {Array<{ ErrType: string, fileName: string, lineContent: string, lineNo: number }>} */
+const errors = [];
+const wikiUrl = 'wiki.fome.tech';
 
-  console.log('Validating absolute URLs...');
+const validateDocRules = (files) => {
+  /**
+   * Check whether links are adhering to custom rules
+   * @param {Array<string>} files - List of file names to check
+  */
+  // ToDo: 0 files have to trigger error! currently incorrect msg below:
+  //  Validating rules for URL links in:  undefined
+  //  ✅ Ok
+  fileMatchIndicator = ""
+  if (files.length > 1){
+      fileMatchIndicator = `${files.length} files`
+  } else {
+      fileMatchIndicator = files[0]
+  }
+  console.log(`Validating rules for URL links in:  ${fileMatchIndicator}`);
+ 
+  // * Check whether links are not staring with "https://wiki.fome.tech"
+  // Note: dynamic javascript string interpolation is used here (see https://www.crstin.com/js-regex-interpolation/)
+  const regexPatternDynAbsLink = new RegExp(`\\]\\((https|http):\\/\\/${wikiUrl}`, 'i');
+  // hint: test static regex via https://regex101.com
+  // * Check whether links are not using a "numbered prefix" like "(/01-blah)"
+  const regexPatternStatNumPrefix = new RegExp(/\(.*\/\d\d\-.*\)/, 'i');
+  // * Check whether links are not markdown links, meaning ending with .md or .mdx like "(/01-blah.md)"
+  const regexPatternStatMdLink = new RegExp(/\(.*\.(md|mdx)\)/, 'i');
 
   files.forEach((fileName) => {
     const lines = fs.readFileSync(fileName, 'utf8').split('\n');
 
     lines.forEach((line) => {
-      if (regexPattern.test(line)) {
+      if (regexPatternDynAbsLink.test(line)) {
+        errors.push({
+          ErrType:"AbsLink"
+          ,fileName
+          ,lineContent: line
+          ,lineNo: lines.indexOf(line) + 1
+        });
+      }
+      if (regexPatternStatMdLink.test(line)) {
         errors.push({
-          fileName,
-          lineContent: line,
-          lineNo: lines.indexOf(line) + 1,
+          ErrType:"MdLink"
+          ,fileName
+          ,lineContent: line
+          ,lineNo: lines.indexOf(line) + 1
+        });
+      }
+      if (regexPatternStatNumPrefix.test(line)) {
+        errors.push({
+          ErrType:"NumPrefix"
+          ,fileName
+          ,lineContent: line
+          ,lineNo: lines.indexOf(line) + 1
         });
       }
     });
   });
 
-  if (errors.length > 0) {
-    console.log('❌ Failed\n');
-    console.log(red(`Absolute URLs to "${wikiUrl}" found in the following files:\n`));
-    errors.forEach((error) => {
-      console.log(
-        `[${error.fileName}:${error.lineNo}] ${error.lineContent.trim()}`
-      );
-    });
-
-    process.exit(1);
-  }
 }
 
 /**
  * Load all md and mdx files from / docs and process them
- */
+*/
 const main = () => {
-  const files = glob.sync('docs/**/*.md?(x)');
+  // ToDo: change in package.json "lint:links": "node scripts/linkValidator.js 'docs/**/*.md?(x)'",
```

Comment:
no need for TODO, you can add it in this PR

Reply:
thats what I forget in initial PR :(